### PR TITLE
Add unique id's and put device id into input_text

### PR DESCRIPTION
**Pull request by woopstar**

### CODE REVIEW

**Comment by heinoskov** [comment]
I have 4 different devices...

Batteries
Battery 1 (I have not used this for anything yet)
Inverter 
Smart Meter (I dont think is used for anything)

hsbo_device_id_inverter:
  name: HSBO Device ID for Huawei Solar Inverter device
hsbo_device_id_batteries:
  name: HSBO Device ID for Huawei Solar Batteries device

File: huawei_solar_battery_optimization.yaml [hunk 1]
```diff
@@ -160,6 +160,8 @@ input_text:
     name: HSBO Recommendation for 16:00
   hsbo_recommended_working_mode:
     name: HSBO Recommended Working Mode
+  hsbo_device_id:
+    name: HSBO Device ID for Huawei Solar device
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:164]: We would need two here for the scripts / huawei actions later in the file. 
  
  hsbo_device_id_inverter:
    name: HSBO Device ID for Huawei Solar device - Inverter
   hsbo_device_id_batteries:
    name: HSBO Device ID for Huawei Solar device - Batteries

File: huawei_solar_battery_optimization.yaml [hunk 2]
```diff
@@ -442,7 +444,7 @@ script:
           periods: |-
             00:01-06:00/1234567/+
             17:00-23:59/1234567/-
-          device_id: 8ee3ed713a270c3bbf9596bcfe414175
+          device_id: "{{ states('input_text.hsbo_device_id') }}"
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:447]: this one should be

device_id: "{{ states('input_text.hsbo_device_id_batteries') }}"

File: huawei_solar_battery_optimization.yaml [hunk 3]
```diff
@@ -502,7 +504,7 @@ script:
       - service: huawei_solar.set_tou_periods
         data:
           periods: 00:00-00:01/1234567/+
-          device_id: 8ee3ed713a270c3bbf9596bcfe414175
+          device_id: "{{ states('input_text.hsbo_device_id') }}"
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:507]: this one should be

device_id: "{{ states('input_text.hsbo_device_id_batteries') }}"

File: huawei_solar_battery_optimization.yaml [hunk 4]
```diff
@@ -528,7 +530,7 @@ script:
     sequence:
       - service: huawei_solar.set_maximum_feed_grid_power_percent
         data:
-          device_id: baba156a5f8b0c02d3503a0e21ce3a8e
+          device_id: "{{ states('input_text.hsbo_device_id') }}"
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:533]: this one should be

device_id: "{{ states('input_text.hsbo_device_id_inverter') }}"

File: huawei_solar_battery_optimization.yaml [hunk 5]
```diff
@@ -546,7 +548,7 @@ script:
...
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:551]: this one should be

device_id: "{{ states('input_text.hsbo_device_id_inverter') }}"

File: huawei_solar_battery_optimization.yaml [hunk 6]
```diff
@@ -971,11 +974,12 @@ sensor:
...
         value_template: >
           {% set target_hour = states('input_number.hsbo_charge_start_time') | int %}
           {% set now_time = now() %}
           {% set target = now_time.replace(hour=target_hour, minute=0, second=0, microsecond=0) %}
           {% if target <= now_time %}
             {% set target = target + timedelta(days=1) %}
           {% endif %}
-          {{ target.strftime('%Y-%m-%d %H:%M:%S') }}
\ No newline at end of file
+          {{ target.strftime('%Y-%m-%d %H:%M:%S') }}
```
Comment (heinoskov) [inline huawei_solar_battery_optimization.yaml:985]: Is there a change here?

**Review by heinoskov — CHANGES_REQUESTED** [review]
I have reviewed and suggested a few changes, mainly as we need multiple devices for the Huawei actions we need for the scripts.

**Comment by woopstar** [comment]
I currently use this lovelace dashboard to validate the HSBO packages, and all the input fields:

```yaml
views:
  - title: HSBO
    path: hsbo
    cards:
      - type: custom:auto-entities
        card:
          type: entities
          title: HSBO Input Fields
        filter:
          include:
            - entity_id: input_text.hsbo*
          exclude: []
      - type: custom:auto-entities
        card:
          type: entities
          title: HSBO Input Fields
        filter:
          include:
            - entity_id: input_number.hsbo*
            - entity_id: input_datetime.hsbo*
          exclude: []
      - type: custom:auto-entities
        card:
          type: entities
          title: HSBO All
        filter:
          include:
            - entity_id: '*hsbo*'
          exclude:
            - entity_id: input_number.hsbo*
            - entity_id: input_datetime.hsbo*
            - entity_id: input_text.hsbo*
```

**Comment by woopstar** [comment]
What are you trying to do here:

`{{ [cutoff_percentage, 100] | min | max(20) | float }}`

The `max(20)` gives an error.

**Comment by heinoskov** [comment]
> What are you trying to do here:
> 
> `{{ [cutoff_percentage, 100] | min | max(20) | float }}`
> 
> The `max(20)` gives an error.

Det skulle have været en min(20) - da AC charge cutoff SOC (%): range går mellem 20-100

**Comment by heinoskov** [comment]
Thanks for your changes! Appreciated! 👍

**Comment by heinoskov** [comment]
> > What are you trying to do here:
> > `{{ [cutoff_percentage, 100] | min | max(20) | float }}`
> > The `max(20)` gives an error.
> 
> Det skulle have været en min(20) - da AC charge cutoff SOC (%): range går mellem 20-100

I have changed this as well

**Pull request merged**